### ISSUE TITLE
[rustdoc] Search results: List exact alias matches before fuzzy ones

### ISSUE THREAD

**Issue by ojeda** [rust-lang/rust#82132]
Aliases should be displayed after non-aliases. Since we have fuzzy search [*], we could match in this order:

- Exact matches (non-aliases)
- Exact matches (aliases)
- Fuzzy search (any, assuming it sorts by some "similarity" index)

If we get #82000, then we could even do:

- Exact matches (non-aliases)
- Exact matches (Rust aliases)
- Exact matches (non-Rust aliases)
- Fuzzy search (any, assuming it sorts by some "similarity" index)

[*]
![image](https://user-images.githubusercontent.com/23638587/107890698-468c4d00-6ee8-11eb-85ed-ea2428ec6d1d.png)

**Comment by lolbinarycat** [comment]
triage: it's not clear what the current and desired behavior is.  also unable to reproduce the list of results shown in the screenshot.

**Comment by ojeda** [comment]
The behavior seems to have changed since then, so the screenshot does not apply anymore.

The linked PR was not merged, so the only thing that may still apply is the first list.

Now, for the first list, it looks harder to reproduce, since we get less fuzzy search results nowadays, apparently. But I tried `writeln`, and I got the exact match first, and then the "fuzzy" ones (removing and even then adding characters) afterwards (e.g. `write`, `write_all`).

So it looks like it now works as this issue asked for, which is great.

Thus closing -- thanks!
